docs(diagnostic): add doc for virtual_text `spacing` and `prefix`

hassec opened the pull request:
I was reading through `vim.diagnostic()` and noticed that these two options weren't yet documented.     

Somewhat related: I did recently stumble over the fact that for each source only the last diagnostic is displayed:    
https://github.com/neovim/neovim/blob/22d7dd2aec9053028cc033e4c68335a81f845e06/runtime/lua/vim/diagnostic.lua#L988-L1004

Would it make sense to explicitly mention this here too? WDYT @gpanders?

hassec committed:
docs(diagnostic): add doc for virtual_text `spacing` and `prefix`

gpanders commented:
Thanks, not sure how that slipped through.

>Somewhat related: I did recently stumble over the fact that for each source only the last diagnostic is displayed:

https://github.com/neovim/neovim/pull/16432 should address this. Adding a blurb in the documentation can't hurt in the meantime though.

hassec committed:
docs(diagnostic): clarify virtual_text behaviour for many diagnostics

Doc now mentions that only one diagnostic message is shown per
namespace, see `vim.diagnostic._get_virt_text_chunks()`

hassec commented:
Thanks for the quick feedback @gpanders :blush:   

I've added a comment about the behavior of `virtual_text` in case of multiple diagnostics.   
For me, it wasn't really the fact whether it is the first or last diagnostic that was surprising, but the fact that only one is shown.       
I'm not saying that is a bad choice, after all, space is limited :man_shrugging: But with the large default prefix of `■` there isn't really any gap between them and all of that lead me to first think: "oh wait something isn't displayed correctly here and the diagnostics overlap"

Anyway, hopefully this brief info in the doc saves someone else from going down that rabbit hole :joy:

dundargoc commented:
Thank you for your contribution! We collect all typo fixes into a single pull request and merge it once it gets big enough: https://github.com/neovim/neovim/pull/16506
